File: dotnet/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/WindowsFactory.cs
namespace AbstractFactoryDesignPattern.Example1;

public class WindowsFactory : GUIFactory
{
	public override Button CreateButton()
	{
		return new WindowsButton();
	}
	public override Menu CreateMenu()
	{
		return new WindowsMenu();
	}
	public override Dialog CreateDialog()
	{
		return new WindowsDialog();
	}
}

File: java/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/src/main/java/creational/abstractfactory/example1/factory/concretes/WindowsFactory.java
package creational.abstractfactory.example1.factory.concretes;

import creational.abstractfactory.example1.factory.abstracts.GUIFactory;
import creational.abstractfactory.example1.models.abstracts.*;
import creational.abstractfactory.example1.models.concretes.*;

public class WindowsFactory extends GUIFactory {
    public Button createButton() { return new WindowsButton(); }
    public Menu   createMenu()   { return new WindowsMenu(); }
    public Dialog createDialog() { return new WindowsDialog(); }
}
